agent/agent.go

```diff
@@ -1560,122 +1558,6 @@ func (a *Agent) segmentConfig() ([]consul.NetworkSegment, error) {
 	return segments, nil
 }
 
-// makeRandomID will generate a random UUID for a node.
-func (a *Agent) makeRandomID() (string, error) {
-	id, err := uuid.GenerateUUID()
-	if err != nil {
-		return "", err
-	}
-
-	a.logger.Debug("Using random ID as node ID", "id", id)
-	return id, nil
-}
-
-// makeNodeID will try to find a host-specific ID, or else will generate a
-// random ID. The returned ID will always be formatted as a GUID. We don't tell
-// the caller whether this ID is random or stable since the consequences are
-// high for us if this changes, so we will persist it either way. This will let
-// gopsutil change implementations without affecting in-place upgrades of nodes.
-func (a *Agent) makeNodeID() (string, error) {
-	// If they've disabled host-based IDs then just make a random one.
-	if a.config.DisableHostNodeID {
-		return a.makeRandomID()
-	}
-
-	// Try to get a stable ID associated with the host itself.
-	info, err := host.Info()
-	if err != nil {
-		a.logger.Debug("Couldn't get a unique ID from the host", "error", err)
-		return a.makeRandomID()
-	}
-
-	// Make sure the host ID parses as a UUID, since we don't have complete
-	// control over this process.
-	id := strings.ToLower(info.HostID)
-	if _, err := uuid.ParseUUID(id); err != nil {
-		a.logger.Debug("Unique ID from host isn't formatted as a UUID",
-			"id", id,
-			"error", err,
-		)
-		return a.makeRandomID()
-	}
```

Comment:
I removed this check, because as of 7f6d95f1c9c67cecf7f3e203813fbaa49a6a5ef8 it should not matter if the HostID is a uuid or not. We hash the string anyway, and build a valid uuid.